core/src/main/scala/org/bitcoins/core/api/dlcoracle/OracleEvent.scala

```diff
@@ -243,9 +243,11 @@ object OracleEvent {
         require(eventDbs.forall(_.attestationOpt.isEmpty),
                 "Cannot have a partially signed event")
 
+        val sortedEventDbs = eventDbs.sortBy(_.nonceIndex)
+
```

Comment:
Is it worth just adding an invariant to `PendingDigitDecompisitionEvent()`  ?  Or in the superclass `OracleEvent`?

Reply:
We can't really because we don't actually save the event dbs or the nonce index, just use the information from the actual eventdbs

Reply:
So my understanding of the purpose of `OracleEvent.fromEventDbs()` is to take in a `Vector[EventDb]` and spit out an in memory respresentation for that event (`OracleEvent`). 

If we add an invariant, for the lifetime of that `OracleEvent` in memory, we KNOW that the nonces are ordered correctly. This patch does that in a similar way, but if we ever have `OracleEvent.fromEventDbs2()` (because we need some new feature or whatever) we need to make sure we propogate this patch here. 

Same thing applies on the oracle explorer side, we need to remember to patch any custom code we have for reading `EventDb`s out of the database -- rather than just having an exception thrown that will remind us during development.

Reply:
> We can't really because we don't actually save the event dbs or the nonce index in the oracle event, just use the information from the actual eventdbs

I think this qualifies as "saving"? It guarantees their in memory representation is ordered

>sortedEventDbs.map(_.nonce),